packages/slate-editor/src/modules/components/InlineContactForm/InlineContactForm.tsx

```diff
@@ -0,0 +1,192 @@
+import type { ContactInfo } from '@prezly/slate-types';
+import classNames from 'classnames';
+import type { FormEvent } from 'react';
+import React from 'react';
+import { useEffect, useState } from 'react';
+
+import { Avatar, Button, Input } from '../../../components';
+
+import styles from './InlineContactForm.module.scss';
+
+interface Props {
+    contact: ContactInfo | null;
+    isEditing?: true;
+    onClose: () => void;
+    onSubmit: (contact: ContactInfo) => void;
+}
+
+const TEXT_MAX_LENGTH = 150;
+
+export function InlineContactForm({ contact, isEditing, onClose, onSubmit }: Props) {
+    const [formState, setFormState] = useState<ContactInfo>({
+        avatar_url: null,
+        address: '',
+        company: '',
+        description: '',
+        email: '',
+        facebook: '',
+        mobile: '',
+        name: '',
+        phone: '',
+        twitter: '',
+        website: '',
+    });
+
+    function handleChange(property: keyof ContactInfo, value: string) {
+        setFormState((formState) => ({
+            ...formState,
+            [property]: value,
+        }));
+    }
+
+    function handleSubmit(event: FormEvent<HTMLFormElement>) {
+        event.preventDefault();
+
+        const isFormValid = event.currentTarget.checkValidity();
+        if (isFormValid) {
+            onSubmit(formState);
+        }
+    }
+
+    useEffect(() => {
+        if (contact) {
+            setFormState(contact);
+        }
+    }, [contact]);
+
+    return (
+        <form onSubmit={handleSubmit}>
```

Comment:
![Screenshot 2023-01-20 at 16 26 07](https://user-images.githubusercontent.com/4209081/213744215-d4f06be3-c56c-4063-85d2-95f8391510ab.png)

The form currently looks like this. I did not add the icons there as we don't have any of those icons (phone, mobile, facebook, twitter) in this repository at the moment.

Do you think it's really important and I should add them or is it fine to skip it? I am a bit lazy here 😄